examples/asynch_adaptive_msm/async_iteration.py

```diff
@@ -72,9 +84,18 @@ async def simulation_iteration(inputs: Iterable[Future[SimulationInput]]):
 
 async def simulation_and_analysis_loop(initial_input):
     simulation_input = initial_input
+    next_conformation = simulation_input.conformation
     model = msmtool.msm_analyzer()
     while not model.is_converged():
-        model, simulation_input = analysis_iteration(model, simulation_iteration(simulation_input))
+        # Option A would combine phase 0 and 1. E.g.:
+        #     simulation = simulation_iteration_optionA(simulation_input, next_conformation)
+        # Option B:
+        # Phase 0: Prepare simulation input with the next round of initial conformations.
+        simulation_input = modify_input(simulation_input, conformation=next_conformation)
+        # Phase 1: Perform simulations
+        simulation = simulation_iteration_optionB(simulation_input)
+        # Phase 2: Update MSM to obtain next set of most informative initial conformations.
+        model, next_conformation = analysis_iteration(model, simulation)
```

Comment:
I do think the paradigm "in an adaptive workflow, the main output of the analysis section is new simulation inputs" is a good paradigm.  I assume that is what A is trying to implement? Option B does read a bit clearer.  For option B, I'd almost put it at the after the analysis iteration to be the most clear to the paradigm, though I'm open to options.

Reply:
Yes. In A, two phases (analysis and simulation) feed directly into each other. In B, three phases comprise simulation, analysis, updated simulation input.

I believe that B is more readable.

I'm not sure what you mean by your last sentence, and it sounds like you are ambivalent about whether you prefer A or B. Can you clarify, or suggest an alternate syntax?